test/cpp/jit/test_op_replacement.cpp

```diff
@@ -0,0 +1,145 @@
+        graph(%0 : Tensor,
+              %1 : Tensor):
+            %2 : Tensor = aten::add(%0, %1)
+            %3 : Tensor  = aten::div(%2, %1)
+            %4 : int = prim::Constant[value=1]()
+            %5: Tensor = aten::_test_serialization_subcmul(%0, %1, %4)
+            return (%3, %5))IR";
+  auto g = std::make_shared<Graph>();
+  test_only_populate_upgraders(test_upgraders);
+  UpgraderEntry test_entry{
+      3,
+      "_test_serialization_subcmul_0_2",
+      "aten::_test_serialization_subcmul(Tensor self, Tensor other, Scalar alpha=2) -> Tensor"};
+  test_only_add_entry("aten::_test_serialization_subcmul", test_entry);
+  torch::jit::parseIR(graph_string, g.get());
+  g->set_op_version(2);
+  ReplaceOldOperatorsWithUpgraders(g);
+  testing::FileCheck()
+      .check("prim::If")
+      ->check_count("aten::div", 2, /*exactly=*/true)
+      ->run(*g);
+  test_only_remove_entry("aten::_test_serialization_subcmul");
+  test_only_remove_upgraders(test_upgraders);
+}
+
+TEST(OpReplacementTest, ReplaceDivInNestedFunction) {
+  const auto graph_string = R"IR(
+        graph(%0 : Tensor,
+              %1 : Tensor,
+              %8 : bool):
+            %9 : bool = prim::Constant[value=1]()
+            %7 : bool = prim::If(%8)
+                block0():
+                    -> (%9)
+                block1():
+                    %2 : Tensor = aten::add(%0, %1)
+                    %3 : Tensor  = aten::div(%2, %1)
+                    %4 : Tensor = aten::add(%3, %0)
+                    %10 : bool = aten::is_floating_point(%4)
+                    -> (%10)
+            return (%7))IR";
+  auto g = std::make_shared<Graph>();
+  test_only_populate_upgraders(test_upgraders);
+  torch::jit::parseIR(graph_string, g.get());
+  g->set_op_version(2);
+  ReplaceOldOperatorsWithUpgraders(g);
+  testing::FileCheck()
+      .check("prim::If")
+      ->check_count("aten::add", 2, false)
+      ->run(*g);
+
+  testing::FileCheck()
+      .check("prim::If")
+      ->check_count("aten::div", 2, false)
+      ->run(*g);
+  test_only_remove_upgraders(test_upgraders);
+}
+
+TEST(OpReplacementTest, ReplaceTestSubcmulInSimpleFunction) {
+  const auto graph_string = R"IR(
+        graph(%0 : Tensor,
+              %1 : Tensor):
+            %3 : int = prim::Constant[value=1]()
+            %2 : Tensor = aten::_test_serialization_subcmul(%0, %1, %3)
+            return (%2))IR";
+  auto g = std::make_shared<Graph>();
+  test_only_populate_upgraders(test_upgraders);
+  UpgraderEntry test_entry{
+      3,
+      "_test_serialization_subcmul_0_2",
+      "aten::_test_serialization_subcmul(Tensor self, Tensor other, Scalar alpha=2) -> Tensor"};
+  test_only_add_entry("aten::_test_serialization_subcmul", test_entry);
+  torch::jit::parseIR(graph_string, g.get());
+  g->set_op_version(2);
+  ReplaceOldOperatorsWithUpgraders(g);
+  testing::FileCheck().check_count("aten::mul", 1, false)->run(*g);
+
+  testing::FileCheck().check_count("aten::sub", 1, false)->run(*g);
```

Comment:
Consider checking `_test_serialization_subcmul` is not present in the graph any more